NE-1071: Default HAProxy maxconn value to 50000 for OCP 4.12

frobware opened the pull request:
This change increases the maximum number of simultaneous connections
that HAProxy will accommodate to 50000. This change will improve
router throughput at the expense of a modest increase in memory usage.

Using the following parameters we see a 3MB increase per-process.

## algorithm=random weight=1 backends=1000 threads=4

```
maxconn  maxconn (HAProxy)  maxsock (HAProxy)  Process Size (MB)
-------  -----------------  -----------------  -----------------
   2000               2000               4054                 53
  20000              20000              40054                 56
  50000              50000             100054                 59
 100000             100000             200054                 66
 200000             200000             400054                 78
   auto             524261            1048576                119
```

And with 64 threads (the current HAProxy maximum) a 4MB increase
per-process:

## algorithm=random weight=1 backends=1000 threads=64

```
maxconn  maxconn (HAProxy)  maxsock (HAProxy)  Process Size (MB)
-------  -----------------  -----------------  -----------------
   2000               2000               4234                 90
  20000              20000              40234                 91
  50000              50000             100234                 95
 100000             100000             200234                101
 200000             200000             400234                114
   auto             524171            1048576                154
```

[1] https://github.com/openshift/enhancements/blob/master/enhancements/ingress/haproxy-max-connections-tuning.md#increased-resource-usage

rfredette commented:
/lgtm

frobware commented:
/retest

Miciah commented:
/approve
/hold
We also need a PR to update <https://github.com/openshift/api/blob/1265e99256880f8679d1b74561c0bc7932067c43/operator/v1/types_ingress.go#L1475-L1483>, and a PR to vendor the update in openshift/cluster-ingress-operator.

Miciah commented:
We'll also need a bug or Jira issue.

frobware commented:
> /approve /hold We also need a PR to update https://github.com/openshift/api/blob/1265e99256880f8679d1b74561c0bc7932067c43/operator/v1/types_ingress.go#L1475-L1483, and a PR to vendor the update in openshift/cluster-ingress-operator.

https://github.com/openshift/api/pull/1299

frobware commented:
> We'll also need a bug or Jira issue.

https://issues.redhat.com/browse/NE-1071

frobware commented:
/retest

frobware commented:
/retest

frobware commented:
/retest

ahardin-rh commented:
/label docs-approved

frobware commented:
@Miciah we have all the labels now - happy to remove your /hold?

frobware commented:
/retest

melvinjoseph86 commented:
/label qe-approved

frobware commented:
@Miciah openshift/api#1299 has now merged and I've updated the cluster-ingress-operator api bump and removed the hold on openshift/cluster-ingress-operator#836.

candita commented:
/refresh jira

candita commented:
/assign @Miciah @candita

Miciah commented:
/hold cancel
